extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -276,14 +278,15 @@ export class AmpStoryPage extends AMP.BaseElement {
 
   /** @override */
   pauseCallback() {
-    this.advancement_.stop();
+    this.timer_.delay((() => {
+      this.advancement_.stop();
+      this.stopListeningToVideoEvents_();
+      this.pauseAllMedia_(true /** rewindToBeginning */);
 
-    this.stopListeningToVideoEvents_();
-    this.pauseAllMedia_(true /** rewindToBeginning */);
-
-    if (this.animationManager_) {
-      this.animationManager_.cancelAll();
-    }
+      if (this.animationManager_) {
+        this.animationManager_.cancelAll();
+      }
+    }), 600);
```

Comment:
This works for desktop, but will have a lot of side effects on mobile: videos on the previous page will keep playing for 600ms.

I believe the right behavior would be to still perform all these operations (stop listening to events, pause the video, cancel the animations...), but only [rewind the video](https://i.ytimg.com/vi/VtLpc_71vtA/maxresdefault.jpg) once the 300ms desktop animation is done (only if desktop?).

Reply:
Okay! I was able to get the delay only on the rewind in the mediapool file if that works so PTAL!